packages/teams-js/src/internal/utils.ts

```diff
@@ -416,9 +416,59 @@ export function validateId(id: string, errorToThrow?: Error): void {
   }
 }
 
-function hasScriptTags(id: string): boolean {
+export function validateUrl(url: URL, errorToThrow?: Error): void {
+  const urlString = url.toString();
+  if (hasScriptTags(urlString)) {
+    throw errorToThrow || new Error('Invalid Url');
+  }
+  if (urlString.length > 2048) {
+    throw errorToThrow || new Error('Url exceeds the maximum size of 2048 characters');
+  }
+  if (!isValidHttpsURL(url)) {
+    throw errorToThrow || new Error('Url should be a valid https url');
+  }
+}
+
+/**
+ * The hasScriptTags function first decodes any HTML entities in the input string using the decodeHTMLEntities function. 
+ * It then tries to decode the result as a URI component. If the URI decoding fails (which would throw an error), it assumes that the input was not encoded and uses the original input.
+ * Next, it defines a regular expression scriptRegex that matches any string that starts with <script (followed by any characters), then has any characters (including newlines),
+ * and ends with </script> (preceded by any characters).
+ * Finally, it uses the test method to check if the decoded input matches this regular expression. The function returns true if a match is found and false otherwise.
+ * @param input URL converted to string to pattern match
+ * @returns true if the input string contains a script tag, false otherwise
+ */
+function hasScriptTags(input: string): boolean {
+  let decodedInput;
+  try {
+    const decodedHTMLInput = decodeHTMLEntities(input);
+    decodedInput = decodeURIComponent(decodedHTMLInput);
+  } catch (e) {
+    // input was not encoded, use it as is
+    decodedInput = input;
+  }
   const scriptRegex = /<script[^>]*>[\s\S]*?<\/script[^>]*>/gi;
-  return scriptRegex.test(id);
+  return scriptRegex.test(decodedInput);
+}
+
+/** 
+ * The decodeHTMLEntities function replaces HTML entities in the input string with their corresponding characters. 
+ * It uses a regular expression to match HTML entities and a replacement function to replace each match with the corresponding character from the entityMap object. 
+ * If a match is not found in the entityMap, it leaves the match as is.
+*/
+function decodeHTMLEntities(input: string): string {
```

Comment:
I understand the regex here but it is really hard to follow, a good alternative would be to:

``` 
function decodeHTMLEntities(input: string): string {
    // Define the entity map
    const entityMap: { [key: string]: string } = {
        '&lt;': '<',
        '&gt;': '>',
        '&amp;': '&',
        '&quot;': '"',
        '&#39;': "'",
        '&#x2F;': '/',
    };
    // Replace each entity with its corresponding value
    for (const entity in entityMap) {
        if (Object.prototype.hasOwnProperty.call(entityMap, entity)) {
            input = input.replace(new RegExp(entity, 'g'), entityMap[entity]);
        }
    } 
    return input;
}

As we won't require to keep track of the Regex here.